connect/transforms/src/main/java/org/apache/kafka/connect/transforms/Cast.java

```diff
@@ -326,7 +337,12 @@ else if (value instanceof String)
     }
 
     private static String castToString(Object value) {
-        return value.toString();
+        if (value instanceof java.util.Date) {
```

Comment:
That makes sense.
One why to do this would be to create an abstraction `o.a.k.connect.data.Logical` which would avoid the whole "if isinstaceof.." code in `Cast` or any other place you're looking to convert to/from logical.